src/vmemcache_repl.c

```diff
@@ -196,17 +200,50 @@ repl_p_lru_new(struct repl_p_head **head)
 
 	util_mutex_init(&h->lock);
 	TAILQ_INIT(&h->first);
+	h->ringbuf = ringbuf_new(LEN_RING_BUF);
 	*head = h;
 
 	return 0;
 }
 
+/*
+ * dequeue_all -- (internal) dequeue all repl_p entries,
+ *                           it MUST be run under a lock
+ */
+static void
+dequeue_all(struct repl_p_head *head)
+{
+	struct repl_p_entry *e;
+	int counter = 0;
+
+	do {
+		e = ringbuf_trydequeue_s(head->ringbuf,
+						sizeof(struct repl_p_entry));
+		if (e == NULL)
+			break;
+
+		TAILQ_MOVE_TO_TAIL(&head->first, e, node);
+		util_bool_compare_and_swap64(e->ptr_entry, NULL, e);
```

Comment:
Why do you even use compare&swap if you are not interested in compare result?

Reply:
It has to be true. I can add assert() here.